docs/docker-and-kubernetes.md

```diff
@@ -0,0 +1,56 @@
+# Docker and Kubernetes
+
+uPortal has a vision of the future that includes pre-packaged ready-to-run
+distributions that can be configured completely at runtime, as you would expect
+any modern cloud-native application to be able to do. In support of this, there
+have been discussions and work done on moving uPortal toward this goal,
+including the building of two Dockerfiles: one container for the portal, and one
+container for the database, whether used directly for non-production, or used to
+initialize an external database using the dumped SQL files.
+
+## Docker build
+
+The docker build process, parameterized and managed primarily via Makefile, is
+as follows:
+
+1. Build the uPortal codebase, specifying a local directory as the build target.
+1. Build an empty database docker image with a few needs for initializing an
+   empty uPortal Postgres database.
+1. Start the previously mentioned database in docker locally to be used for
+   the uPortal init process
+1. Query the docker daemon for the IP address of the locally-running database,
+   and use this information to create a temporary
+   filters/docker-build.properties file, inserting the IP address.
+1. Use ant to initialize the database into the running docker postgres
+   container.
+1. Commit the database container, and run the Docker build for the uPortal
+   container.
+1. Push both containers to the named REGISTRY
```

Comment:
See https://github.com/Jasig/uPortal/pull/813#discussion_r101292214 for the previous discussion on markdown order lists.

Outcome from previous discussion

> Write for readability. Use the style that includes the numbers in the documentation source.